### DISCUSSION TITLE
Undisabling the HTTPING Formula, help needed

### DISCUSSION THREAD

**Discussion by iandol** [Homebrew/discussions#3837] [Writing Formulae/Casks]
The httping formula in core has been disabled due to the repo being removed:

https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/httping.rb
https://github.com/Homebrew/homebrew-core/commit/9780370a01db5d215d63448b82edfaa4baf263c0

However the original author does have a github repo and I confirmed with him this is the current maintained source. NOTE: I am no a C programmer at all and I am confused over basic compilation issues. So I downloaded the master and tried to build it, but hit a number of things I needed to mod:

* I hit an error due to `useconds_t` being undefined even though `#include <unistd.h>` is present in `utils.c`; this is some posix library and I have no idea what to do, I solved it by replacing `useconds_t` with `unsigned int` in the code itself, but maybe there is a better way to do this?:
```
In file included from nc.c:22:
./utils.h:19:15: error: unknown type name 'useconds_t'
void myusleep(useconds_t v);
```
* Some files in the makefile are renamed in the archive, like `readme.txt` to `README.md`, is `inreplace` OK for this?
* `openssl@1.1` is a named dependency. For plain building of the master I adjusted `LDFLAGS` / `CFLAGS` to point to it, but in the formula `def install` I see this doesn't happen, what is the magic to make this so?

**Comment by iandol** [comment]
I can succesfully build and install with this modified formula `def install`:

```ruby
  def install
    # Reported upstream: https://github.com/folkertvanheusden/HTTPing/issues/5
    inreplace "utils.h", "useconds_t", "unsigned int"
    # Reported upstream, see: https://github.com/Homebrew/homebrew/pull/28653
    inreplace %w[configure Makefile], "lncursesw", "lncurses"
    ENV.append "LDFLAGS", "-lintl" if OS.mac?
    inreplace "Makefile", "cp nl.mo $(DESTDIR)/$(PREFIX)/share/locale/nl/LC_MESSAGES/httping.mo", ""
    # Reported upstream: https://github.com/folkertvanheusden/HTTPing/issues/6
    inreplace "Makefile", "license.txt license.OpenSSL readme.txt", "LICENSE README.md"
    system "make", "install", "PREFIX=#{prefix}"
  end
```

Does this look OK?

For OpenSSL it seems my answer is brew does include the paths for me, no need to manually tweak this...

**Reply by SMillerDev** [reply]
It's hard to tell without CI and the rest of the formula, please make a pull request.

**Reply by iandol** [reply]
> It's hard to tell without CI and the rest of the formula, please make a pull request.

OK, thanks for the feedback!

**Comment by iandol** [comment]
OK, here is the pull request: https://github.com/Homebrew/homebrew-core/pull/114088
